src/icp_canister_backend/src/lib.rs

```diff
@@ -62,54 +69,103 @@ pub async fn deposit(user: Principal, timelock: u64) -> Result<(), types::Deposi
         }
     }).ok_or(types::DepositError::LedgerNotSet)?;
 
-    let deposit_key = format!("{}:{}", user.to_text(), timelock);
-
-    let exists = DEPOSITS.with(|deposits| {
-        deposits.borrow().contains_key(&deposit_key)
-    });
-
-    if exists {
-        return Err(types::DepositError::DepositAlreadyExists);
-    }
-    
-    // Generate subaccount and check balance
+    // Generate subaccount
     let subaccount = get_deposit_subaccount(user, timelock);
-    let account = types::Account {
+    let from_account = types::Account {
         owner: ic_cdk::api::id(),
         subaccount: Some(subaccount.to_vec()),
     };
 
+    // Check balance in subaccount
     let balance_result: Result<(Nat,), _> = call(
         ledger_principal,
         "icrc1_balance_of",
-        (account,)
+        (from_account.clone(),)
     ).await;
 
     let balance = match balance_result {
         Ok((balance,)) => balance,
         Err(_) => return Err(types::DepositError::LedgerCallFailed),
     };
 
-    let balance_u64: u64 = balance.0.try_into().map_err(|_| types::DepositError::InternalError)?;
+    let balance_u64: u64 = balance.0.clone().try_into().map_err(|_| types::DepositError::InternalError)?;
 
     if balance_u64 == 0 {
```

Comment:
You check if balance > then fee later in the code, so you don't need this check here.